actionpack/lib/action_dispatch/journey/router.rb

```diff
@@ -27,45 +27,20 @@ def eager_load!
       end
 
       def serve(req)
-        find_routes(req) do |match, parameters, route|
-          set_params  = req.path_parameters
-          path_info   = req.path_info
-          script_name = req.script_name
-
-          unless route.path.anchored
-            req.script_name = (script_name.to_s + match.to_s).chomp("/")
-            req.path_info = match.post_match
-            req.path_info = "/" + req.path_info unless req.path_info.start_with? "/"
-          end
-
+        find_routes(req) do |route, parameters|
           req.path_parameters = parameters
           req.route = route
 
           _, headers, _ = response = route.app.serve(req)
 
-          if headers[Constants::X_CASCADE] == "pass"
-            req.script_name     = script_name
-            req.path_info       = path_info
-            req.path_parameters = set_params
-            next
-          end
-
-          return response
+          return response unless headers[Constants::X_CASCADE] == "pass"
         end
 
         [404, { Constants::X_CASCADE => "pass" }, ["Not Found"]]
       end
 
-      def recognize(rails_req)
-        find_routes(rails_req) do |match, parameters, route|
-          unless route.path.anchored
-            rails_req.script_name = match.to_s
-            rails_req.path_info   = match.post_match
-            rails_req.path_info   = "/" + rails_req.path_info unless rails_req.path_info.start_with? "/"
-          end
-
-          yield(route, parameters)
-        end
+      def recognize(rails_req, &block)
+        find_routes(rails_req, &block)
       end
```

Comment:
Why not just use an alias? Is the method even still needed?

Reply:
I wanted to keep the diff clean, but I agree that an alias (or inlining everything into `recognize`) makes sense to me.

I'll update with another commit